Linux-Application/DomesdayDuplicator/advancednamingdialog.cpp

```diff
@@ -180,6 +196,56 @@ void AdvancedNamingDialog::updateGui(void)
 
 }
 
+// Update the GUI and hold values from previous side input
+void AdvancedNamingDialog::updateSideHoldings(void)
+{
+    if ((ui->discSideSpinBox->value()) == 2 ) {
+
+        if (mintSide1.length() == 0){
+        mintSide1 = ui->mintLineEdit->text();
+        }
+        if (notesSide1.length() == 0) {
+        notesSide1 = ui->notesLineEdit->text();
+        }
+
+        ui->mintLineEdit->setText(mintSide2);
+        ui->notesLineEdit->setText(notesSide2);
+    }
+
+    if ((ui->discSideSpinBox->value()) == 1) {
+
+        if (mintSide2.length() == 0){
+        mintSide2 = ui->mintLineEdit->text();
+        }
+        if (notesSide2.length() == 0){
+        notesSide2 = ui->notesLineEdit->text();
+        }
+
+        ui->mintLineEdit->setText(mintSide1);
+        ui->notesLineEdit->setText(notesSide1);
+    }
+
+    if ((ui->discSideSpinBox->value()) == 3) {
+
+        if (mintSide2.length() == 0){
+        mintSide2 = ui->mintLineEdit->text();
+        }
+        if (notesSide2.length() == 0){
+        notesSide2 = ui->notesLineEdit->text();
+        }
+
+        ui->mintLineEdit->setText("");
+        ui->notesLineEdit->setText("");
+    }
+
+    if ((ui->discSideSpinBox->value() != 1 and ui->discSideSpinBox->value() != 2 and ui->discSideSpinBox->value() != 3 )) {
+
+        ui->mintLineEdit->setText("");
+        ui->notesLineEdit->setText("");
+    }
+
+}
+
```

Comment:
Given you have special cases here for sides 1, 2 and 3, I'd wonder if the code should be allowing comments for any number of sides? That is, whatever you're doing to split up the comments at the moment should actually split into an arbitrary length list, so you can enter mint marks for a 4-sided capture etc. - then the variables would be arrays rather than foo1 and foo2.